src/OperationalInsights/OperationalInsights/Models/PSCluster.cs

```diff
@@ -67,7 +67,11 @@ public PSCluster(Cluster cluster)
             this.IsAvailabilityZonesEnabled = cluster.IsAvailabilityZonesEnabled;
             this.BillingType = cluster.BillingType;
 
-            if (cluster.KeyVaultProperties != null)
+            if (cluster.KeyVaultProperties == null || PSKeyVaultProperties.CreateProperties(cluster.KeyVaultProperties.KeyVaultUri, cluster.KeyVaultProperties.KeyName, cluster.KeyVaultProperties.KeyVersion) == null)
+            {
+                this.KeyVaultProperties = null;
+            }
+            else
             {
                 this.KeyVaultProperties = new PSKeyVaultProperties(cluster.KeyVaultProperties);
```

Comment:
```suggestion
                this.KeyVaultProperties = PSKeyVaultProperties.CreateProperties(cluster.KeyVaultProperties.KeyVaultUri, cluster.KeyVaultProperties.KeyName, cluster.KeyVaultProperties.KeyVersion)
```

Reply:
this constructor serves the response received from the calls to the OperationalInsights Nuget (OperationalInsightsClient.Clusters.cs). 
the response contains these 3 properties and also KeyRsaSize which cannot be passed from PSH.
the CreateProperties function serves the cmdlets(such that KeyRsaSize is null) - NewAzureOperationalInsightsClusterCommand.cs, UpdateAzureOperationalInsightsClusterCommand.cs